src/material/list/list-base.ts

```diff
@@ -282,6 +282,9 @@ export abstract class MatListItemBase implements AfterViewInit, OnDestroy, Rippl
       const treatAsTitle = this._titles.length === 0 && numberOfLines === 1;
       unscopedContentEl.classList.toggle('mdc-list-item__primary-text', treatAsTitle);
       unscopedContentEl.classList.toggle('mdc-list-item__secondary-text', !treatAsTitle);
+      if (unscopedContentEl.textContent) {
+        unscopedContentEl.textContent = unscopedContentEl.textContent.trim();
```

Comment:
Thx for the PR. I don't think this is the right solution. There might be other nodes part of the unscoped content, bindings etc.

Usually consumers can avoid the whitespace by using a span, but maybe it can also be improved with a better CSS white space setting?

Reply:
I couldn't get whitespace property of CSS to impact in this case. It looks like the problem is due to the use of `inline-block` on the before pseudo element `.mdc-list-item__secondary-text:before`. This is added by the material-components library. I find the same issue on their [demo website](https://material-components.github.io/material-components-web-catalog/#/component/list)  in case there is some unscoped content.

[MDN](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace#spaces_in_between_inline_and_inline-block_elements) suggests to use flexbox or using negative margin.

Reply:
It sounds like we should work with the MDC base styles to fix this issue.